src/plugins/ucx/ucx_backend.cpp

```diff
@@ -584,25 +576,22 @@ nixlUcxEngine::nixlUcxEngine (const nixlBackendInitParams* init_params)
                                           numWorkers,
                                           init_params->syncMode);
 
-    for (unsigned int i = 0; i < numWorkers; i++)
+    for (size_t i = 0; i < numWorkers; i++) {
         uws.emplace_back(std::make_unique<nixlUcxWorker>(*uc, err_handling_mode));
-
-    const auto &uw = uws.front();
-    workerAddr = uw->epAddr();
-
-    if (workerAddr.empty()) {
-        NIXL_ERROR << "Failed to get UCX worker address";
-        initErr = true;
-        return;
     }
 
+    workerAddr = uws.front()->epAddr();
+
     if (pthrOn) {
         for (auto &uw: uws) {
             pollFds.push_back({uw->getEfd(), POLLIN, 0});
         }
         pollFds.push_back({pthrControlPipe[0], POLLIN, 0});
     }
 
+    // TODO: in case of UCX error handling is enabled, we can clean up AM based connections error
+    //       handling, if user requested disabled error handling, we dont care about it.
+    auto &uw = uws.front();
```

Comment:
You can also revert this change to return the line before 583.

Reply:
I  mean, previously it was:
```C++
const auto &uw = uws.front();
workerAddr = uw->epAddr();
...
uw->regAmCallback(CONN_CHECK, connectionCheckAmCb, this);
```
, no it's:
```
workerAddr = uws.front()->epAddr();
...
auto &uw = uws.front();
uw->regAmCallback(CONN_CHECK, connectionCheckAmCb, this);
```

And you can revert this change, since it's not needed.

Reply:
yes, the intention of this change is to remove `uw` variable together with AM callbacks.